repositories/productRepository.go

```diff
@@ -202,6 +226,17 @@ func (repository *productRepository) UpdateProduct(product models.Product) error
 	if commandTag != 1 {
 		return utils.NoDataError
 	}
+	message := make(map[string]interface{})
```

Comment:
When you're creating a collection with make, it's better ( if you already know ) to set the size of it, like:
message := make(map[string]interface{}, 2)
This way there will be avoided unnecessary resource holds and potential memory leaks.